data/prepare_dataset.py

```diff
@@ -0,0 +1,90 @@
+import os
+
+import fortepyan as ff
+from tqdm import tqdm
+from quantizer import MidiQuantizer
+from datasets import Value, Dataset, Features, Sequence, DatasetDict, load_dataset
+
+
+def process_dataset(dataset_path: str, split: str, sequence_len: int, quantizer: MidiQuantizer) -> list[dict]:
+    dataset = load_dataset(dataset_path, split=split)
+
+    processed_records = []
+
+    for record in tqdm(dataset, total=dataset.num_rows):
+        # print(record)
+        piece = ff.MidiPiece.from_huggingface(record)
+        processed_record = process_record(piece, sequence_len, quantizer)
+
+        processed_records += processed_record
+
+    return processed_records
+
+
+def process_record(piece: ff.MidiPiece, sequence_len: int, quantizer: MidiQuantizer) -> list[dict]:
+    piece_quantized = quantizer.quantize_piece(piece)
+
+    midi_filename = piece_quantized.source["midi_filename"]
+
+    record = []
+
+    for subset in tqdm(piece_quantized.df.rolling(window=sequence_len, step=sequence_len)):
+        # rolling sometimes creates subsets with shorter sequence length, they are filtered here
+        if len(subset) != sequence_len:
+            continue
+
+        sequence = {
+            "midi_filename": midi_filename,
+            "pitch": subset.pitch.astype("int16").values.T,
+            "dstart_bin": subset.dstart_bin.astype("int16").values.T,
+            "duration_bin": subset.duration_bin.astype("int16").values.T,
+            "velocity_bin": subset.velocity_bin.astype("int16").values.T,
+        }
+
+        record.append(sequence)
+
+    return record
+
+
+if __name__ == "__main__":
+    # get huggingface token from environment variables
+    token = os.environ["HUGGINGFACE_TOKEN"]
+
+    # hyperparameters
+    sequence_len = 128
+
+    # That's where I'm downloading the LTAFDB data
+    hf_dataset_path = "roszcz/maestro-v1"
```

Comment:
```suggestion
    hf_dataset_path = "roszcz/maestro-v1-sustain"
```

This is a version of the dataset with the sustain pedal applied. This makes the data much closer to the original performance.